src/mqueryfront/src/QueryPage.js

```diff
@@ -90,10 +91,60 @@ class QueryPage extends Component {
         
     }
 
+
+    loadJob() {
+        const LIMIT = 50;
+
+        if (!this.state.qhash) {
+            return;
+        }
+
+        axios
+            .get(
+                API_URL +
+                    "/matches/" +
+                    this.state.qhash +
+                    "?offset=" +
+                    0 +
+                    "&limit=" +
+                    LIMIT
+            )
+            .then((response) => {
+                let newShouldRequest = true;
+
+                if (
```

Comment:
maybe move setState above this, and return in if, instead of using a temprary vraiable here?